complex objects/cargo/list/src/lib.rs

```diff
@@ -43,108 +41,125 @@ impl ListManager {
-        db.execute(sql, &[&name]).unwrap();
-        self.fetch_category(&name)
+        db.execute(sql, &[&name, &color]).unwrap();
+        self.fetch_label(&name)
     }
 
-    pub fn fetch_category(&self, name: &String) -> Option<Category> {
-        let sql = r#"SELECT id, name FROM categories WHERE name=?"#;
+    pub fn fetch_label(&self, name: &String) -> Option<Label> {
+        let sql = r#"SELECT name, color FROM labels WHERE name=?"#;
 
         let conn = self.store.read_connection();
         let mut stmt = conn.prepare(sql).unwrap();
-        let mut category_iter = stmt.query_map(&[name], |row| {
-            Category {
-                id: row.get(0),
-                name: row.get(1),
-                items: Vec::new()
+        let mut label_iter = stmt.query_map(&[name], |row| {
+            Label {
+                name: row.get(0),
+                color: row.get(1)
             }
         }).unwrap();
 
-        if let Some(result) = category_iter.next() {
-            if let Some(mut category) = result.ok() {
-                category.items = self.fetch_items_for_category(&category);
-                Some(category)
-            } else {
-                None
-            }
+        if let Some(result) = label_iter.next() {
+            result.ok()
         } else {
-            println!("No category found");
+            println!("No label found for name {:?}", name);
             None
         }
     }
 
-    pub fn fetch_categories(&self) -> Vec<Category> {
-        let sql = r#"SELECT id, name
-                     FROM categories"#;
+    pub fn fetch_labels(&self) -> Vec<Label> {
+        let sql = r#"SELECT name, color
+                     FROM labels"#;
         let conn = self.store.read_connection();
         let mut stmt = conn.prepare(sql).unwrap();
-        let mut category_iter = stmt.query_map(&[], |row| {
-            Category {
-                id: row.get(0),
-                name: row.get(1),
-                items: Vec::new()
+        let mut label_iter = stmt.query_map(&[], |row| {
+            Label {
+                name: row.get(0),
+                color: row.get(1)
             }
         }).unwrap();
 
-        let mut category_list: Vec<Category> = Vec::new();
-        while let Some(result) = category_iter.next() {
-            if let Some(mut category) = result.ok() {
-                category.items = self.fetch_items_for_category(&category);
-                category_list.push(category);
+        let mut label_list: Vec<Label> = Vec::new();
+        while let Some(result) = label_iter.next() {
+            if let Some(mut label) = result.ok() {
+                label_list.push(label);
             }
         }
-        category_list
+        label_list
+    }
+
+    pub fn fetch_labels_for_item(&self, item_uuid: &String) -> Vec<Label> {
+        let sql = r#"SELECT name, color
+                     FROM labels LEFT JOIN item_labels on item_labels.item_uuid=?"#;
```

Comment:
I think this should just be a join, not a left join.

Reply:
Otherwise you'd get labels assigned to item uuid, and all the other labels as well.

Reply:
Doh! Perils of doing things quickly.